plugins/chat/assets/javascripts/discourse/components/chat/drawer-routes/members.gjs

```diff
@@ -0,0 +1,67 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import didUpdate from "@ember/render-modifiers/modifiers/did-update";
+import { service } from "@ember/service";
+import I18n from "discourse-i18n";
+import Navbar from "discourse/plugins/chat/discourse/components/chat/navbar";
+import ChannelMembers from "discourse/plugins/chat/discourse/components/chat/routes/channel-info-members";
+import ChannelInfoNav from "discourse/plugins/chat/discourse/components/chat/routes/channel-info-nav";
+
+export default class ChatDrawerRoutesMembers extends Component {
+  @service chat;
+  @service chatStateManager;
+  @service chatChannelsManager;
+
+  activeChannel = null;
+
+  get backButton() {
+    return {
+      route: "chat.channel",
+      models: this.chat.activeChannel?.routeModels,
+      title: I18n.t("chat.return_to_channel"),
+    };
+  }
+
+  @action
+  fetchChannel() {
+    if (!this.args.params?.channelId) {
+      return;
+    }
+
+    return this.chatChannelsManager
+      .find(this.args.params.channelId)
+      .then((channel) => {
+        this.chat.activeChannel = channel;
+      });
+  }
+
+  <template>
+    <Navbar @onClick={{this.chat.toggleDrawer}} as |navbar|>
+      <navbar.BackButton
+        @title={{this.backButton.title}}
+        @route={{this.backButton.route}}
+        @routeModels={{this.backButton.models}}
+      />
+      <navbar.ChannelTitle @channel={{this.chat.activeChannel}} />
+      <navbar.Actions as |a|>
+        <a.ToggleDrawerButton />
+        <a.FullPageButton />
+        <a.CloseDrawerButton />
+      </navbar.Actions>
+    </Navbar>
+
+    {{#if this.chatStateManager.isDrawerExpanded}}
+      <div
+        class="chat-drawer-content"
+        {{didInsert this.fetchChannel}}
```

Comment:
What does call `this.fetchChannel` do when called without any arguments/parameters? 🤔 

Looks like we're doing

```js
    if (!this.args.params?.channelId) {
       return;
     }
```

Reply:
The channel id param should always exist here due to how we extract it from the url within chat drawer routes. I think this guard clause is would only catch instances of badly formatted url. Changing this to async should mean we can remove the didUpdate and simplify things.